controllers/observability_controller.go

```diff
@@ -226,20 +227,21 @@ func (r *ObservabilityReconciler) InitializeOperand(mgr ctrl.Manager) error {
 	}
 
 	storageClassExists := false
-	storageClass := &storage.StorageClass{}
-	selector := client.ObjectKey{
-		Name: storageClassGp2,
+	cluster := client.ObjectKey{
+		Name: "cluster",
 	}
+	infra := &infrastructure.Infrastructure{}
 
-	err = apiReader.Get(context.Background(), selector, storageClass)
-	if err == nil {
-		// no error means that the storage class was found
+	err = r.Get(context.Background(), cluster, infra)
+
+	// Check the infrastructure, If its Libvirt that means it is running on crc so it will run without storage.
+	if err == nil && string(infra.Status.PlatformStatus.Type) != "Libvirt" {
```

Comment:
Maybe we could something like:

```suggestion
	if err == nil && (string(infra.Status.PlatformStatus.Type) != string(infrastructure.LibvirtPlatformType) || string(infra.Status.PlatformStatus.Type) != string(infrastructure.NonePlatformType) )  {
		storageClassExists = true
	}
```